### DISCUSSION TITLE
SadTalker install on termux android

### DISCUSSION THREAD

**Discussion by Modernpharmacist94** [OpenTalker/SadTalker#590] [General]
So i tried to install miniconda on termux on Samsung galaxy s22 5g but i couldn't run the installer as an exe,i tried bard ai to download and install SadTalker and it didn't work i asked this question How to install SadTalker from vinthony on termux? git clone https://github.com/vinthony/sadtalker.git then this command pip3 install -r requirements.txt it gave me an error  pip subprocess to install backend dependencies did not run successfully. , then i decided to follow the exact commands vinthony provided for linux installation the first main issue i can't excute miniconda via command bash Miniconda3-latest-Linux-arm64.sh it says no such file or directory i'm in the third command conda create -n sadtalker python=3.8 can't install conda yet!!! Help !!

**Comment by DaveTheDayTrader** [comment]
Why on earth would you try to use a smartphone ? lol

When I was totally bored I played with Andronix / termux ( I have laptops. desktops, tablets, phones , MACs, Chrombooks etc. tried everything) 

Are you trying this just as a challenge ? lol

Im using a cheap old Dell T3500  

![Screenshot_2023-09-24_21-39-49](https://github.com/OpenTalker/SadTalker/assets/145946551/af38259c-d9c6-4408-8c96-9f40b8f1e4c4)

Install Automatic 1111 local and play with the Sadtalker extension

**Reply by Modernpharmacist94** [reply]
I don't have laptop or PC ,i work in two full time jobs and i want to start content creation to make an extra money that's why i'm trying to install it in Termux

**Reply by Modernpharmacist94** [reply]
Download Termux from F-droid website
1. Termux-setup-storage (then allow access to storage)
2. apt update
3. apt upgrade
4. pkg update
5. pkg upgrade
6. pkg install proot-distro
7. proot-distro install ubuntu
8. proot-distro login ubuntu
9. su -
10. apt-get update
11. apt-get install sudo
12. sudo apt install python3
13. sudo apt install python3-pip
14. sudo apt-get install python-is-python3
15. sudo apt-get install git
16. apt-get install wget
17. wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-aarch64.sh
Note : my device architecture is aarch64 but maybe your device is x86-64
18. sha256sum Miniconda3-latest-Linux-aarch64.sh
19. chmod +x Miniconda3-latest-Linux-aarch64.sh
20. ./Miniconda3-latest-Linux-aarch64.sh
Follow the installation prompt and install
21. conda config --set auto_activate_base false
22. git clone https://github.com/OpenTalker/SadTalker.git
23. cd SadTalker
24. conda create -n sadtalker python=3.8
25. conda activate sadtalker
26. pip install torch==1.12.1+cu113 torchvision==0.13.1+cu113 torchaudio==0.12.1 --extra-index-url https://download.pytorch.org/whl/cu113 (error here)
27. conda install ffmpeg
28. pip install -r requirements.txt
29. pip install TTS (error here)
30. bash scripts/download_models.sh
31. bash webui.sh (error here)

**Reply by arun9706** [reply]
Bro did it work ? I want to install miniconda on my tablet and could do it because of this .
